app/src/components/dialog/EditDialog.tsx

```diff
@@ -0,0 +1,106 @@
+import { Box, Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle } from '@material-ui/core';
+import React from 'react';
+import { Formik } from 'formik';
+
+export interface IEditDialogComponentProps {
+  element: any;
+  initialValues: any;
+  validationSchema: any;
+}
+
+export interface IEditDialogProps {
+  /**
+   * The dialog window title text.
+   *
+   * @type {string}
+   * @memberof IEditDialogProps
+   */
+  dialogTitle: string;
+  /**
+   * The dialog window body text.
+   *
+   * @type {string}
+   * @memberof IEditDialogProps
+   */
+  dialogText: string;
+  /**
+   * Set to `true` to open the dialog, `false` to close the dialog.
+   *
+   * @type {boolean}
+   * @memberof IEditDialogProps
+   */
+  open: boolean;
+
+  /**
+   * @type {IEditDialogComponentProps}
+   * @memberof IEditDialogProps
+   */
+  component: IEditDialogComponentProps;
+
+  /**
+   * Callback fired if the dialog is closed.
+   *
+   * @memberof IEditDialogProps
+   */
+  onClose: () => void;
+  /**
+   * Callback fired if the 'No' button is clicked.
+   *
+   * @memberof IEditDialogProps
+   */
+  onCancel: () => void;
+  /**
+   * Callback fired if the 'Yes' button is clicked.
+   *
+   * @memberof IEditDialogProps
+   */
+  onSave: (values: IEditDialogComponentProps) => void;
+  //TODO
+  //Add interface for the values
+}
+
+/**
+ * A dialog for displaying a title + message (typically a question), and giving the user the option to say
```

Comment:
this comment probably needs to be updated (seems like it is referring to the yes/no dialog)

Reply:
ok